### PR TITLE
Reproducibility docs

### PR DESCRIPTION
**Pull request by rokroskar**
This PR adds a page discussing reproducibility in Renku. 

Note: to build, make sure to use the included `Pipfile`. This uses a fork of `sphinx` which fixes a bug in the sphinx graphviz extension. To build the docs:

```
$ pipenv install  --dev --deploy
$ rm -rf docs/_build/ && sphinx-build -nN docs docs/_build/html/
```
Then open `docs/_build/html/index.html`

---

Addresses #300

### CODE REVIEW

**Comment by jirikuncar** [comment]
@rokroskar for testing it's better to use `$ pipenv install --deploy`

**Comment by 3C111** [comment]
I build the docs as per instructions, but the images do not show up completely.

**Comment by rokroskar** [comment]
sorry my instructions were slightly broken -- can you do `pipenv shell` before building the docs?

**Comment by 3C111** [comment]
I did that, else it wouldn't work at all.

**Comment by rokroskar** [comment]
@3C111 sorry I had another typo I fixed now... it should be:

```
$ pipenv install --dev --deploy 
```

**Comment by 3C111** [comment]
Now: 
> Running Sphinx v1.8.0+
And the images appeared!

**Comment by rokroskar** [comment]
@3C111 what would you add at the end?

**Comment by 3C111** [comment]
I read this single page only, so that influences the flow of the document. Maybe a concluding sentence tying these things together. Or a short summary about how Renku will take care of these things if you don't want to be bothered by it, or so.

File: docs/user/reproducibility.rst [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+* How exactly was this model or result obtained? How was the raw data pre-processed?
+* Who is using my data, model, or result and how?
+* What does this new data mean for our last month's report?
+* How can I share my latest analysis with a colleague in the cloud?
+* How can I make sure that our colleagues in another team can reproduce our results exactly?
+
+Answering these questions in a reliable fashion is difficult. Renku provides
+the tools that substantially reduce the effort required on the part of the
+data scientist who wants to keep her work reproducible, reliable and robust.
+Here, we introduce the essential concepts underyling the design and
```
Comment (3C111) [inline docs/user/reproducibility.rst]: And spotted a typo when re-reading "underyling" -> "underlying"

Comment (rokroskar) [reply docs/user/reproducibility.rst]: thanks, I forgot to run the spellcheck... there were a few others

**Comment by rokroskar** [comment]
@3C111 absolutely, there needs to be a doc that will tie all of this together. Maybe this doc can finish with a set of pointers to docs describing the actual pieces of the platform that deal with these three issues, e.g. something like:

```
Renku helps you achieve the goal of fully reproducible data science by bundling together several technologies. 

* for keeping track of the lineage, our CLI relies heavily on the Common Workflow Language. [Here](link-to-cwl-page) you can learn more about our CWL integration.
* for version control, we rely on `git`
* your project includes a container specification from day 1. Check out [these docs](link-to-CI-docs) to learn more about how we build images for your project and what we do with them

**Comment by 3C111** [comment]
Yes, that sounds good!

**Comment by rokroskar** [comment]
@3C111 I'm assigning you to this for now :) Could you please implement the three bullet points in the comment above and make stubs for those pages? They can be empty with a "work in progress". We can then make issues to fill them up.

File: docs/user/reproducibility.rst [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+* How exactly was this model or result obtained? How was the raw data pre-processed?
+* Who is using my data, model, or result and how?
+* What does this new data mean for our last month's report?
+* How can I share my latest analysis with a colleague in the cloud?
+* How can I make sure that our colleagues in another team can reproduce our results exactly?
+
+Answering these questions in a reliable fashion is difficult. Renku provides
+the tools that substantially reduce the effort required on the part of the
+data scientist who wants to keep her work reproducible, reliable and robust.
```
Comment (leafty) [inline docs/user/reproducibility.rst]: --> [...] required on the part of data scientist**s** who wants to keep **their** work reproducible [...]

**Pull request merged**